packages/react-dom-bindings/src/server/ReactDOMServerExternalRuntime.js

```diff
@@ -13,13 +14,70 @@ import {
   completeSegment,
 } from './fizz-instruction-set/ReactDOMFizzInstructionSet';
 
-// Intentionally does nothing. Implementation will be added in future PR.
-// eslint-disable-next-line no-unused-vars
-const observer = new MutationObserver(mutations => {
-  // These are only called so I can check what the module output looks like. The
-  // code is unreachable.
-  clientRenderBoundary();
-  completeBoundaryWithStyles();
-  completeBoundary();
-  completeSegment();
-});
+if (document.readyState === 'loading') {
+  if (!window.$RC) {
+    // TODO: Eventually remove, we currently need to set these globals for
+    // compatibility with ReactDOMFizzInstructionSet
+    window.$RC = completeBoundary;
+    window.$RM = new Map();
+  }
+  const mutationObserver = new MutationObserver(mutations => {
+    for (let i = 0; i < mutations.length; i++) {
+      const addedNodes = mutations[i].addedNodes;
+      for (let j = 0; j < addedNodes.length; j++) {
+        if (addedNodes.item(j).parentNode) {
+          handleNode(addedNodes.item(j));
+        }
+      }
+    }
+  });
+
+  // We assume that instruction data nodes are eventually appended to the
+  // body, even if Fizz is streaming to a shell / subtree.
+
+  // $FlowFixMe[incompatible-cast] document.body should exist at this point
+  mutationObserver.observe((document.body /*: Node */), {
+    childList: true,
+  });
```

Comment:
if watching `documentElement` does not work perhaps check if body exists and if not observe `documentElement` until the body shows up and then instantiate this observation?